cmd/kops/export_kubecfg.go

```diff
@@ -45,6 +45,9 @@ var (
 
 	# export using a user already existing in the kubeconfig file
 	kops export kubecfg kubernetes-cluster.example.com --user my-oidc-user
+
+	# export using the internal DNS name which bypasses the cloud load balancer
+	kops export kubecfg kubernetes-cluster.example.com --internal
```

Comment:
Maybe add a quick note here that you need an additional SG?

Reply:
Or maybe should add by default 443 when `sslCertificateId` is enabled and export "internal"?

Reply:
I think that is a good idea!
Non-internal endpoints doesn't work anyway with this setup, right?

Reply:
the problem with modifying security groups is that `kops export kubecfg` is decoupled from where SG definitions are managed in `kops update cluster`. At `update cluster` time we don't know if the cluster operator will ever be running `kops export kubecfg --internal`. while we do generate the kubecfg in `update cluster --yes`, I suppose we could add the security group rule if the `--internal` flag is passed to `update cluster` but that could lead to confusion. Users running `kops export kubecfg --internal` would need to know to include `--internal` in their `update cluster` commands as well. I don't know if we have precedent for making a "pair" of commands require similar flags like that.

Reply:
Can we just add the port if sslCertificateId is enabled?
The export part can be explained in docs or printed at update cluster summary.

Reply:
I think always adding the rule if sslCertificateId is present might be reasonable. It assumes that anyone using an ACM cert will want the ability to use client cert auth for api access. I'm failing to think of scenarios in which that wouldn't be the case. I haven't looked into the code but I'm assuming we can use the same sources (CIDRs or SG IDs) we use on the API ELB for the instance 443 as well. I don't see a need to split that out into its own flag or diverge those settings.

One could go a step further and imply the `--internal` flag's behavior when exporting kubecfg if `sslCertificateId` and `--admin` are set. In fact 1.19 might be the best time to do that because the --admin flag is being added, so we could document this change in behavior along side the new flag.

Reply:
👍

Reply:
I could see a site with an alternate authentication provider not wanting the ability to use client cert auth from outside the cluster. But I see little reason to restrict anything that can hit the API ELB from hitting the instance's port 443.

I agree that `sslCertificateId` and `--admin` should imply the `--internal` flag's behavior.

Reply:
After more thought, creating the additional SG rules when `kops update cluster --internal` would be problematic for organizations in which multiple people run their kops commands manually. If someone were to forget to include `--internal` then kops would revoke the SG rules.

It would be much preferable to add the SG rules based on fields in the ClusterSpec. Either `sslCertificateId` being set or we add an additional field that allows users to toggle the behavior. Which we choose depends on if we think anyone with `sslCertifiateId` set would _not_ want the SG rules added.

Regardless, I think adding `--internal` while also implying `--internal` when `sslCertificateId` and `--admin` are used is a good idea.